lxd/device/tpm.go

```diff
@@ -217,13 +217,25 @@ func (d *tpm) startVM() (*deviceConfig.RunConfig, error) {
 		},
 	}
 
-	proc, err := subprocess.NewProcess("swtpm", []string{"socket", "--tpm2", "--tpmstate", fmt.Sprintf("dir=%s", tpmDevPath), "--ctrl", fmt.Sprintf("type=unixio,path=%s", socketPath)}, "", "")
+	// Method to handle paths longer than 108 characters.
```

Comment:
Would it make sense to export `shortenedSocketPath()` in a way so that it can also be used here?

Reply:
yes, it would work the same way. But are you suggesting I disassociate the function from the `qemu` type? And if so how would I handle the `d.addFileDescriptor` call inside `shortenedSocketPath`?